File: Backend/backend-sgac/src/main/java/org/uteq/sgacfinal/entity/TipoEstadoRegistro.java
package org.uteq.sgacfinal.entity;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@Entity
@Table(name = "tipo_estado_registro", schema = "ayudantia")
public class TipoEstadoRegistro {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id_tipo_estado_registro", nullable = false)
    private Integer id;

    @Size(max = 50)
    @NotNull
    @Column(name = "nombre_estado", nullable = false, length = 50)
    private String nombreEstado;

    @Column(name = "descripcion", length = Integer.MAX_VALUE)
    private String descripcion;

    @Column(name = "activo")
    private Boolean activo;

}

File: Frontend/frontend-sgac/src/app/core/models/catalogos/TipoEstadoRegistro.ts
export interface TipoEstadoRegistroResponse {
  id: number;
  nombre_estado: string;
  descripcion: string;
  codigo: string;
  activo: boolean;
}

export interface TipoEstadoRegistroRequest {
  nombreEstado: string;
  descripcion: string;
  codigo: string;
}
